config/taxonomy/HOWTO_SPEC.md

```diff
@@ -0,0 +1,276 @@
+# Writing taxonomy specification
+
+The project uses schema definitions to define taxonomy.
+This document lists the requirements for writing such schemas.
+
+## What spec to follow?
+
+We use a subset of json schema DRAFT 4 that is described in this document. 
+
+You SHOULD declare the schema version in the document:
+
+```json
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    ...
+}
+```
+
+The goal is to preserve compatibility with json schema DRAFT 4, OpenAPI 3.0, and Kubernetes CRDs (all are based on json schema DRAFT 4), and code generation tools.
+We hope that newer json schema specs will be supported in the future as we improve our tooling.
+
+
+## Schema properties
+
+In this section we describe all schema properties that can be used. We categorize the properties to [structural properties](#structural-properties) and [validation properties](#validation-properties).
+
+### Structural properties
+
+The structural properties define the schema objects in terms of fields and types (cf. validation). They are the core of defining a
+[_structural schema_](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#specifying-a-structural-schema): a schema without polymorphism.
+
+The following is a list of structural properties that can be used:
+
+| Keyword | Type	| Description | Recommendation
+| --- | --- | --- | ---
+| type  | string | The type the schema: string, number, integer, boolean, array, or object. | MUST
+| properties | object | Defines properties of the schema. Each value of this object MUST be a schema object. | MUST if defining a non-empty struct
+| additionalProperties | bool or object | See [Field `additionalProperties`](#field-additionalproperties) | MUST if defining a map. SHOULD otherwise.
+| items | object | Defines the schema of items in an array  | MUST if `type` is array
+| description | string | Provide explanation about the purpose of the instance described by this schema. | SHOULD
+| title | string | A short description | MAY
+| default | value of `type` | Supply a default JSON value | MAY
+
+The fields `properties`, `additionalProperties` with object type, and `items` are mutually exclusive:
+1. Use `properties` to define object fields
+    ```json
+    // object with name and age
+    {
+        "type": "object",
+        "properties": {
+            "name": {
+                "type": "string"
+            },
+            "age": {
+                "type": "integer",
+            }
+        }
+    }   
+    ```
+2. Use `additionalProperties` with object type to define a map
+    ```json
+    // map from string to integer values
+    {
+        "type": "object",
+        "additionalProperties": {
+            "type": "integer"
+        }
+    }
+    ```
+3. Use `items` to define the type of array items
+    ```json
+    // array of strings
+    {
+        "type": "array",
+        "items": {
+            "type": "string"
+        }
+    }
```

Comment:
When creating an array of geography items I had to define it slightly differently in order for the required name property in geography to be enforced.  "items": { "properties": { "geography": {"$ref": "#/definitions/geography"}}}

Reply:
You definition should be fixed to:
```json
    {
        "type": "array",
        "items": {
            "type": "object",
            "properties": {
              "geography": {
                "$ref": "#/definitions/geography"
              }
            }
        }
    }
```

It's the same as here but defining an object rather than a string as the array item type.